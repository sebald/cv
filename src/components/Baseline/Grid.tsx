import * as React from 'react';
import { createPortal } from 'react-dom';

import { Baseline } from './Baseline';
import { BaselineGridState } from './types';

export class BaselineGrid extends React.Component<{}, BaselineGridState> {
  el: HTMLElement;
  state: BaselineGridState = {
    show: false,
  };

  constructor(props: object) {
    super(props);
    this.el = document.createElement('div');
  }

  handleKeyPress = (ev: KeyboardEvent) => {
    const modifier = ev.metaKey || ev.ctrlKey;
    if (modifier && ev.key === 'b') {
      this.setState(prevState => ({ show: !prevState.show }));
    }
  };

  componentDidMount() {
    document.body.appendChild(this.el);
    window.addEventListener('keypress', this.handleKeyPress);
  }

  componentWillUnmount() {
    document.body.removeChild(this.el);
    window.removeEventListener('keypress', this.handleKeyPress);
  }

  render() {
    return this.state.show ? createPortal(<Baseline />, this.el) : null;
  }
}
