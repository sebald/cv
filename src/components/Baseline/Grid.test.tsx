import { mount } from 'enzyme';
import * as React from 'react';

import { BaselineGrid } from './Grid';

test('render nothing', () => {
  const wrapper = mount(<BaselineGrid />);
  expect(wrapper).toMatchSnapshot();
});

test('render <SysmetricBaseline/>', () => {
  const wrapper = mount(<BaselineGrid />);
  wrapper.setState({ show: true });
  expect(wrapper).toMatchSnapshot();
});
