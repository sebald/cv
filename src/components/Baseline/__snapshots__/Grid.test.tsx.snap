// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <SysmetricBaseline/> 1`] = `
.c0 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: -1;
  background-image: repeating-linear-gradient( to bottom, rgba(255,105,180,0.5), rgba(255,105,180,0.5) 1px, transparent 1px, transparent NaNrem );
}

<BaselineGrid>
  <Baseline>
    <div
      className="c0"
    />
  </Baseline>
</BaselineGrid>
`;

exports[`render nothing 1`] = `<BaselineGrid />`;
